docs/content/docs/dev/configuration/maven.md

```diff
@@ -0,0 +1,175 @@
+
+## Adding dependencies to the project
+
+Open the `pom.xml` file in your profile directory and add the dependency in between
+the `dependencies` tab.  
+
+For example, you can add the Kafka connector as a dependency like this:
+
+```xml
+<dependencies>
+    
+    <dependency>
+        <groupId>org.apache.flink</groupId>
+        <artifactId>flink-connector-kafka</artifactId>
+        <version>{{< version >}}</version>
+    </dependency>
+    
+</dependencies>
+```
+
+Then execute `mvn install` on the command line. 
+
+Projects created from the `Java Project Template`, the `Scala Project Template`, or Gradle are configured
+to automatically include the application dependencies into the application JAR when you run `mvn clean package`.
+For projects that are not set up from those templates, we recommend adding the Maven Shade Plugin to
+build the application jar with all required dependencies.
+
+**Important:** Note that all these core API dependencies should have their scope set to [*provided*](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope). This means that
+they are needed to compile against, but that they should not be packaged into the project's resulting
+application JAR file. If not set to *provided*, the best case scenario is that the resulting JAR
+becomes excessively large, because it also contains all Flink core dependencies. The worst case scenario
+is that the Flink core dependencies that are added to the application's JAR file clash with some of
+your own dependency versions (which is normally avoided through inverted classloading).
+
+To correctly package the dependencies into the application JAR, the Flink API dependencies must 
+be set to the *compile* scope.
+
+## Packaging the application
+
+Depending on your use case, you may need to package your Flink application in different ways before it
+gets deployed to a Flink environment. 
+
+If you want to create a JAR for a Flink Job and use only Flink dependencies without any third-party 
+dependencies (i.e. using the filesystem connector with JSON format), you do not need to create an 
+uber/fat JAR or shade any dependencies.
+
+If you want to create a JAR for a Flink Job and use external dependencies not built into the Flink 
+distribution, you can either add them to the classpath of the distribution or shade them into your 
+uber/fat application JAR.
+
+With the generated uber/fat JAR, you can submit it to a local or remote cluster with:
+
+```sh
+bin/flink run -c org.example.MyJob myFatJar.jar
+```
+
+To learn more about how to deploy Flink jobs, check out the [deployment guide]({{< ref "docs/deployment/cli" >}}).
+
+## Template for creating an uber/fat JAR with dependencies
+
+To build an application JAR that contains all dependencies required for declared connectors and libraries,
+you can use the following shade plugin definition:
+
+```xml
```

Comment:
Overall I find it weird that we don't just link to the quickstarts. They are ultimately the source of truth of how a recommended project looks like, and are also tested.

Reply:
I think this page is more for detailed/advanced topics for maven. This is a bit duplicated but I think it should be synced with what we have in the quickstarts. We can also improve it later after this PR.

Reply:
After some discussion, I think we can improve it after the PR.